2.0.0: fails to build with avahi support disabled (--disable-avahi)

**Describe the bug**

```
$ make -j1
gdbus-codegen \
    --interface-prefix org.cups.cupsd \
    --c-namespace Cups \
    --generate-c-code cups-notifier \
    daemon/org.cups.cupsd.Notifier.xml
make  all-recursive
make[1]: Verzeichnis „/home/domain/user/Downloads/cups-browsed-2.0.0“ wird betreten
make[2]: Verzeichnis „/home/domain/user/Downloads/cups-browsed-2.0.0“ wird betreten
  CC       backend/implicitclass-implicitclass.o
  CCLD     implicitclass
  CC       daemon/cups_browsed-cups-browsed.o
daemon/cups-browsed.c:405:3: error: unknown type name 'AvahiIfIndex'
  405 |   AvahiIfIndex interface;
      |   ^~~~~~~~~~~~
daemon/cups-browsed.c:406:3: error: unknown type name 'AvahiProtocol'
  406 |   AvahiProtocol protocol;
      |   ^~~~~~~~~~~~~
daemon/cups-browsed.c:411:9: error: unknown type name 'AvahiAddress'
  411 |   const AvahiAddress *address;
      |         ^~~~~~~~~~~~
daemon/cups-browsed.c:413:3: error: unknown type name 'AvahiStringList'
  413 |   AvahiStringList *txt;
      |   ^~~~~~~~~~~~~~~
daemon/cups-browsed.c:414:3: error: unknown type name 'AvahiLookupResultFlags'
  414 |   AvahiLookupResultFlags flags;
      |   ^~~~~~~~~~~~~~~~~~~~~~
daemon/cups-browsed.c: In function 'get_local_queue_name':
daemon/cups-browsed.c:6196:13: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
 6196 |         str = resource;
      |             ^
daemon/cups-browsed.c: In function 'examine_discovered_printer_record':
daemon/cups-browsed.c:10199:7: warning: unused variable 'raw_queue' [-Wunused-variable]
10199 |   int raw_queue = 0;
      |       ^~~~~~~~~
daemon/cups-browsed.c: At top level:
daemon/cups-browsed.c:3897:1: warning: 'local_printer_has_uuid' defined but not used [-Wunused-function]
 3897 | local_printer_has_uuid (gpointer key,
      | ^~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:1075: daemon/cups_browsed-cups-browsed.o] Fehler 1
make[2]: Verzeichnis „/home/domain/user/Downloads/cups-browsed-2.0.0“ wird verlassen
make[1]: *** [Makefile:1260: all-recursive] Fehler 1
make[1]: Verzeichnis „/home/domain/user/Downloads/cups-browsed-2.0.0“ wird verlassen
make: *** [Makefile:769: all] Fehler 2
```

**To Reproduce**
Steps to reproduce the behavior:
1. ./configure --disable-avahi
2. make
3. See error

**Expected behavior**
Allow to build with avahi support disabled.

**Screenshots**
-

**System Information:**
 - OS: Linux
 - Version -

**Additional context**
-

--- a/daemon/cups-browsed.c
+++ b/daemon/cups-browsed.c
@@ -400,6 +400,7 @@ typedef struct pagesize_count_s
   int   count;
 } pagesize_count_t;
 
+#ifdef HAVE_AVAHI
 typedef struct resolver_args_s
 {
   AvahiIfIndex interface;
@@ -414,6 +415,7 @@ typedef struct resolver_args_s
   AvahiLookupResultFlags flags;
   void* userdata;
 } resolver_args_t;
+#endif // HAVE_AVAHI
 
 typedef struct create_args_s
 {

